skypy/halo/mass.py

```diff
@@ -1,38 +1,319 @@
+    collapse, see equation 10 in [1]_ or [2]_.
+
+    Parameters
+    -----------
+    sigma: (ns,) array_like
+        Array of the mass variance at different scales and at a given redshift.
+    params: float
+        The :math:`{A,a,p, delta_c}` parameters of the Sheth-Tormen formalism.
+
+    Returns
+    --------
+    f_c: array_like
+        Array with the values of the collapse function.
+
+    Examples
+    ---------
+    >>> import numpy as np
+    >>> from skypy.halo import mass
+    >>> from skypy.power_spectrum import _eisenstein_hu as eh
+    >>> from skypy.power_spectrum import growth_function
+
+    This example will compute the Press-Schecter function for
+    spherical collapse and a Planck15 cosmology. The power spectrum is
+    given by the Eisenstein and Hu fitting formula:
+
+    >>> from astropy.cosmology import Planck15
+    >>> cosmo = Planck15
+    >>> k = np.logspace(-3, 1, num=5, base=10.0)
+    >>> A_s, n_s = 2.1982e-09, 0.969453
+    >>> Pk = eh.eisenstein_hu(k, A_s, n_s, cosmo, kwmap=0.02, wiggle=True)
+
+    The Sheth-Tormen collapse function at redshift 0:
+
+    >>> m = 10**np.arange(9.0, 12.0, 2)
+    >>> sigma = np.sqrt(_sigma_squared(m, k, Pk, 0, cosmo))
+    >>> mass.sheth_tormen_collapse_function(sigma,
+    ...     params=(0.3222, 0.707, 0.3, 1.686))
+    array([0.17947815, 0.19952375])
+
+    And the Press-Schechter collapse function at redshift 0:
+
+    >>> mass.press_schechter_collapse_function(sigma)
+    array([0.17896132, 0.21613726])
+
+    References
+    ----------
+    .. [1] R. K. Sheth and G. Tormen,  Mon. Not. Roy. Astron. Soc. 308, 119
+        (1999), astro-ph/9901122.
+    .. [2] https://www.slac.stanford.edu/econf/C070730/talks/
+        Wechsler_080207.pdf
+    '''
+    A, a, p, delta_c = params
+
+    return A * np.sqrt(2.0 * a / np.pi) * (delta_c / sigma) * \
+        np.exp(- 0.5 * a * (delta_c / sigma)**2) * \
+        (1.0 + np.power(np.power(sigma / delta_c, 2.0) / a, p))
+
+
+press_schechter_collapse_function = partial(sheth_tormen_collapse_function,
+                                            params=(0.5, 1, 0, 1.69))
+press_schechter_mass_function = partial(
+                                halo_mass_function,
+                                collapse_function=sheth_tormen_collapse_function,
+                                params=(0.5, 1, 0, 1.69))
+press_schechter_mass_sampler = partial(halo_mass_sampler,
+                                       mass_function=halo_mass_function,
+                                       params=(sheth_tormen_collapse_function,
+                                               (0.5, 1, 0, 1.69)))
+
+
+def _sigma_squared(M, wavenumber, linear_power_today,
+                   redshift, cosmology):
+    k = wavenumber
+    Pk = linear_power_today
+    M = np.atleast_1d(M)[:, np.newaxis]
+
+    # Growth function
+    D0 = growth_function(0, cosmology)
+    Dz = growth_function(redshift, cosmology) / D0
```

Comment:
Pass Dz as an argument to halo_mass_sampler instead of redshift (I don't think redshift is used anywhere else?) That way people are free to use whichever model for the growth function they prefer and aren't restricted to specifically `skypy.power_spectrum.growth_function`